cmd/otelcol/main.go

```diff
@@ -306,6 +307,26 @@ func setMemoryLimit(memTotalSizeMiB int) {
 	log.Printf("Set memory limit to %d MiB", memLimit)
 }
 
+// Sets environment variables expected by agent_config.yaml if missing
+func setDefaultEnvVars() {
+	var tmpVar string
+	realm := os.Getenv("SPLUNK_REALM")
+	token := os.Getenv("SPLUNK_ACCESS_TOKEN")
+	testArgs := [][]string{
+		{"SPLUNK_API_URL", "https://api." + realm + ".signalfx.com"},
+		{"SPLUNK_INGEST_URL", "https://ingest." + realm + ".signalfx.com"},
+		{"SPLUNK_TRACE_URL", "https://ingest." + realm + ".signalfx.com/v2/trace"},
+		{"SPLUNK_HEC_URL", "https://ingest." + realm + ".signalfx.com/v1/log"},
+		{"SPLUNK_HEC_TOKEN", token},
+	}
+	for _, v := range testArgs {
+		tmpVar = os.Getenv(v[0])
```

Comment:
This would prevent setting empty values, which could be desired for testing or other edge cases.  Would it make sense to use `LookupEnv()` and only set if not in the current env?

Reply:
Hmm, that does seem to make sense -- let me take a look